### ISSUE TITLE
Both "Reset stored target" and "Store Target" on the same attack doesn't work

### ISSUE THREAD

**Issue by rversteegen** [ohrrpgce/ohrrpgce#42]
"Reset stored target" happens after "Store Target", so if you set both bits on the same attack, Store Target does nothing. But being able to set both bits would be very useful, as it makes Store Target work the way you might expect it does (*set* the stored target rather than adding to the list of stored target mask).

It's not enough to swap the order the bits take effect, because `inflict()` is called multiple times if there are multiple targets. Instead Reset stored target needs to happen before the first `inflict()` call.

**Comment by bob-the-hamster** [comment]
I completely agree. The current sequence of operations is just the result of my not thinking things out enough when I first implemented this

**Comment by rversteegen** [comment]
This bug is also related to #1099  "Spread absorb attacks don't display total amount absorbed" which also involves moving code out of inflict so that it doesn't happen multiple times for multiple targets. Might as well fix both at once.
